frontend/src/framework/internal/EnsembleSetLoader.ts

```diff
@@ -0,0 +1,126 @@
+import { EnsembleDetails, EnsembleParameterDescription, EnsembleSensitivity } from "@api";
+import { apiService } from "@framework/ApiService";
+import { QueryClient } from "@tanstack/react-query";
+
+import { Ensemble, Sensitivity, SensitivityCase, SensitivityType } from "../Ensemble";
+import { EnsembleIdent } from "../EnsembleIdent";
+import { EnsembleSet } from "../EnsembleSet";
+
+export async function loadEnsembleSetMetadataFromBackend(
+    queryClient: QueryClient,
+    ensemblesToLoad: EnsembleIdent[]
+): Promise<EnsembleSet> {
+    console.debug("loadEnsembleSetMetadataFromBackend", ensemblesToLoad);
+
+    const STALE_TIME = 5 * 60 * 1000;
+    const CACHE_TIME = 5 * 60 * 1000;
+    // const STALE_TIME = 0;
+    // const CACHE_TIME = 0;
+
+    const ensembleDetailsPromiseArr: Promise<EnsembleDetails>[] = [];
+    const sensitivityPromiseArr: Promise<EnsembleSensitivity[]>[] = [];
+    const parametersPromiseArr: Promise<EnsembleParameterDescription[]>[] = [];
+
+    for (let i = 0; i < ensemblesToLoad.length; i++) {
+        const caseUuid = ensemblesToLoad[i].getCaseUuid();
+        const ensembleName = ensemblesToLoad[i].getEnsembleName();
+
+        const ensembleDetailsPromise = queryClient.fetchQuery({
+            queryKey: ["getEnsembleDetails", caseUuid, ensembleName],
+            queryFn: () => apiService.explore.getEnsembleDetails(caseUuid, ensembleName),
+            staleTime: STALE_TIME,
+            cacheTime: CACHE_TIME,
+        });
+        ensembleDetailsPromiseArr.push(ensembleDetailsPromise);
+
+        const sensitivityPromise = queryClient.fetchQuery({
+            queryKey: ["getSensitivities", caseUuid, ensembleName],
+            queryFn: () => apiService.parameters.getSensitivities(caseUuid, ensembleName),
+            staleTime: STALE_TIME,
+            cacheTime: CACHE_TIME,
+        });
+        sensitivityPromiseArr.push(sensitivityPromise);
+
+        const parametersPromise = queryClient.fetchQuery({
+            queryKey: ["getParameterNamesAndDescription", caseUuid, ensembleName],
+            queryFn: () => apiService.parameters.getParameterNamesAndDescription(caseUuid, ensembleName),
+            staleTime: STALE_TIME,
+            cacheTime: CACHE_TIME,
+        });
+        parametersPromiseArr.push(parametersPromise);
+
+        console.debug("Issued promises:", i);
```

Comment:
If implementing `for ... of` this could be `console.debug("Issued promises:", parametersPromiseArr.length)`.
If not, this should probably be `i+1`?